### PR TITLE
Bug 1210304 - Fix some David's comments about more apps navigation

### PR DESCRIPTION
**Pull request by MariaKolesina**
More apps navigation has been changed
Comments for all functions in more_apps_navigation.js have been added.

### CODE REVIEW

**Comment by MariaKolesina** [comment]
@harman-red-square/harman-reviewers

File: apps/verticalhome/js/more_apps_navigation.js [hunk 1]
```diff
@@ -2,84 +2,103 @@
 (function (exports) {
 
   var MoreAppsNavigation = {
-    controls: null,
+    items: null,
     selectedElemIndex: 0,
-    selectElement: null,
 
+    /**
+     * Initialize list of navigation in MoreAppsNavigation when all items are
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: list of navigation -> navigation list? or am i misunderstanding this sentence?

File: apps/verticalhome/js/more_apps_navigation.js [hunk 2]
```diff
@@ -2,84 +2,103 @@
...
+     * loaded
+     */
     init: function () {
-      this.controls = app.grid.getItems();
-      this.reset();
+      this.items = app.grid.getItems();
     },
 
+    /**
+     * Called when 'More Apps' screen has been opened. This function reset
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: function resets selected element to a first element and adds...

File: apps/verticalhome/js/more_apps_navigation.js [hunk 3]
```diff
@@ -2,84 +2,103 @@
...
-      this.selectElement = this.controls[this.selectedElemIndex].element;
-      this.selectElement.classList.add('selected');
-      this.clearNavigationEvents();
+      this.items[this.selectedElemIndex].element.classList.add('selected');
       window.addEventListener('keydown', this);
     },
 
-    clearNavigationEvents: function () {
+    /**
+     * Remove listener of keydown event, when "More Apps" screen is hide
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: I think it is better to move "when "More Apps" screen is hidden" under something like "Note:" section in a comment.
So, overall description here might be:
"Removes listener of the keydown event.
Note: called when "More Apps" screen is closed"

File: apps/verticalhome/js/more_apps_navigation.js [hunk 4]
```diff
@@ -2,84 +2,103 @@
...
+     */
+    stopListeningKeydownEvents: function () {
       window.removeEventListener('keydown', this);
     },
 
+    /**
+     * Handle keydown event on 'More Apps' screen
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: *Handles

File: apps/verticalhome/js/more_apps_navigation.js [hunk 5]
```diff
@@ -2,84 +2,103 @@
...
         case 'Accept':
-          window.removeEventListener('keydown', this);
-          this.selectElement.click();
-          break;
-        case 'BrowserBack':
-        case 'Backspace':
-          app.hideMoreApps();
+          /**
+           * Launch the selected app. Don't need to call hideMoreApps(), because
+           * it is called when the app will close
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: "when the app will close" -> "when app is closed"

File: apps/verticalhome/js/more_apps_navigation.js [hunk 6]
```diff
@@ -2,84 +2,103 @@
...
     },
 
-    changeSelectElem: function (deltaIndex) {
-      this.selectElement.classList.remove('selected');
-      var temp = this.selectedElemIndex + deltaIndex;
-      if (temp > -1 && temp < this.controls.length) {
-        this.selectedElemIndex = temp;
+    /**
+     * Change old selected item to new selected item and scroll it into view if
+     * it doesn't fully visible. If new selectedElemIndex is output of 'items'
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: output of 'items' -> out of 'items' array range

File: apps/verticalhome/js/more_apps_navigation.js [hunk 7]
```diff
@@ -2,84 +2,103 @@
...
+     * array limits, then current selected item don't change
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: then current selected item don't change -> then currently selected item remains unchanged

File: apps/verticalhome/js/more_apps_navigation.js [hunk 8]
```diff
@@ -2,84 +2,103 @@
...
+     */
+    changeSelectElem: function (deltaIndex, behavior) {
+      var trialIndex = this.selectedElemIndex + deltaIndex;
+      if (trialIndex > -1 && trialIndex < this.items.length) {
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: minor.
I personally prefer to use "x >= 0 &&  x < size" if statements. Feels a bit more readable for me.

File: apps/verticalhome/js/more_apps_navigation.js [hunk 9]
```diff
@@ -2,84 +2,103 @@
...
+                  scrollIntoView({behavior: 'smooth', block: 'start'});
+        }
       }
-      this.selectElement = this.controls[this.selectedElemIndex].element;
-      this.selectElement.classList.add('selected');
     },
 
-    isVisible: function (element) {
+    /**
+     * Check that selected element is fully visible in the screen
```
Comment (sergkashin) [inline apps/verticalhome/js/more_apps_navigation.js]: in -> on

**Comment by sergkashin** [comment]
looks good to me with a few minor docs comments

**Comment by MariaKolesina** [comment]
Thank you, all comments fixed

**Comment by ruvsmirnov** [comment]
It looks good for me, too.

**Pull request merged**